src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx

```diff
@@ -2,3 +2,572 @@ export const meta = {
+
+//Fetch a single record by its identifier
+const oneTodo = await API.graphql({
+  query: queries.getTodo,
+  variables: { id: 'some id' }
+});
+```
+
+</Block>
+
+</BlockSwitcher>
+
+<Accordion title='Troubleshooting deny-by-default behavior' headingLevel='4' eyebrow='Troubleshooting'>
+
+You can run into unauthorized errors if you do not have at least one auth rule defined. 
+
+Use the `@auth` directive to configure authorization rules for public, sign-in user, per user, and per user group data access. **Authorization rules operate on the deny-by-default principle**. Meaning that if an authorization rule is not specifically configured, it is denied.
+
+```graphql
+type Todo @model @auth(rules: [{ allow: owner }]) {
+  content: String
+}
+```
+
+In the example above, each signed-in user, or also known as "owner", of a Todo can create, read, update, and delete their own Todos.
+
+Amplify also allows you to restrict the allowed operations, combine multiple authorization rules, and apply fine-grained field-level authorization.
+
+``` graphql
+type Todo @model @auth(rules: [
+  { allow: public, operations: [read]},
+  { allow: owner }
+]) {
+  content: String
+}
+```
+
+In the example above, everyone (`public`) can read every Todo but owner (authenticated users) can create, read, update, and delete their own Todos.
+
+To learn more, see the [Authorization Rules for the API](https://docs.amplify.aws/cli/graphql/authorization-rules/).
+
+</Accordion>
+
+<br/>
+
+<Accordion title='Custom authorization mode to query data' headingLevel='4' eyebrow='Learn more'>
+
+Each AWS AppSync API uses a default authorization mode when you configure your app. To override this default, pass an `authMode` property. For example, this is useful when you have public reads through API Key auth and authenticated reads through IAM auth. The following examples show how you can query data with custom authorization mode:
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { API } from 'aws-amplify';
+import { GraphQLQuery, GRAPHQL_AUTH_MODE } from '@aws-amplify/api';
+import * as queries from './graphql/queries';
+import { GetTodoQuery } from './API';
+
+const todos = await API.graphql<GraphQLQuery<GetTodoQuery>>({
+  query: queries.listTodos,
+  authMode: GRAPHQL_AUTH_MODE.AWS_IAM
+});
+```
+
+</Block>
+
+<Block name="JavaScript">
+
+```js
+import { API } from 'aws-amplify';
+import { GRAPHQL_AUTH_MODE } from '@aws-amplify/api';
+import * as queries from './graphql/queries';
+
+
+const todos = await API.graphql({
+  query: queries.listTodos,
+  authMode: GRAPHQL_AUTH_MODE.AWS_IAM
+});
+```
+
+</Block>
+</BlockSwitcher>
+
+</Accordion>
+
+## Paginate your data
+  
+As your data grows, you will want to filter and paginate your list queries. Fortunately, this is already built in to the Amplify-generated GraphQL API.
+
+You can find the input schemas in the Docs pane of the GraphQL explorer or inside your autogenerated `/graphql` folder. They look like this:
```

Comment:
@renebrandel Line 162 TODO Follow up on note "the content below here might be a bit too technical as the starting point. (and also duplicative with paginate section lower down.)"